sdk/nodejs/yaml/yaml.ts

```diff
@@ -120,14 +120,10 @@ import * as outputs from "../types/output";
         resourcePrefix?: string;
     }
 
-    function yamlLoadAll(text: string): any[] {
-        // NOTE[pulumi-kubernetes#501]: Use `loadAll` with `JSON_SCHEMA` here instead of
-        // `safeLoadAll` because the latter is incompatible with `JSON_SCHEMA`. It is
-        // important to use `JSON_SCHEMA` here because the fields of the Kubernetes core
-        // API types are all tagged with `json:`, and they don't deal very well with things
-        // like dates.
-        const jsyaml = require("js-yaml");
-        return jsyaml.loadAll(text, undefined, {schema: jsyaml.JSON_SCHEMA});
+    function yamlLoadAll(text: string): pulumi.Output<any[]> {
+        const promise = pulumi.output(pulumi.runtime.invoke(
+            "kubernetes:yaml:decode", {text}, {async: true}));
+        return promise.apply<any[]>(p => p.result);
```

Comment:
Note that I don't think there is any need/value in wrapping in `pulumi.output` - can just use the promise returned from `invoke` as-is (and `.then` instead of `.apply`).  Not a big deal - but just probably unnecessary.  Also - `Input` accepts either a `Promise` or an `Output` .